### DISCUSSION TITLE
Could be possible to make osu lazer on uwp format for testing in xbox dev mode?

### DISCUSSION THREAD

**Discussion by JLiveOf** [ppy/osu#27658] [Ideas] [answered]
my question is if it is possible to compile osu on uwp to test it in the xbox dev mode, i have a very janky pc but what if osu lazer had a uwp install format for an xbox series only to test if connecting a keyboard and mouse on the xbox works and its actually playable?

**Comment by JLiveOf** [comment]
i would be happy if i could test an osu uwp on my xbox series via the developer mode cause to play it smoothly on my pc i have to set the screen resolution to 800x600 px and in the xbox dev mode you can sideload appx files without the needing of publishing the game to a store

**Comment by Rekunan** [comment, marked as answer]
No plans to officially support UWP (#23600) so you'll have to manage it on your own, keeping in mind that it won't be able to connect to official servers due to compatibility and anticheat concerns.

From some quick searching online, it's possible, but you might have to make some adjustments to the codebase. I def recommend looking through the [UWP docs](https://docs.microsoft.com/windows/uwp) for this. In particular, they have a page for [general conversion/migration to UWP](https://docs.microsoft.com/windows/uwp/porting/desktop-to-uwp-root), and also [xbox-specific pages](https://docs.microsoft.com/windows/uwp/xbox-apps/index).
